include/splitvector/splitvec.h

```diff
@@ -899,16 +916,16 @@ class SplitVector {
     * @param newSize The new size of the SplitVector.
     */
    DEVICEONLY
-   void device_resize(size_t newSize, bool construct = true) {
+   void device_resize(size_t newSize,bool _construct=true) {
       if (newSize > capacity()) {
          assert(0 && "Splitvector has a catastrophic failure trying to resize on device.");
       }
-      if (construct) {
+      if constexpr (!std::is_trivially_copy_constructible_v<T> ){
```

Comment:
Yes! A vector has its contents intialized as zero by default. Being able to skip that initialization is something I use in the GPU code if I'm going to write into that buffer in a moment, but it should not be default behaviour, as it would be very different from std::vector then.